### DISCUSSION TITLE
SlimSelect Events not working

### DISCUSSION THREAD

**Discussion by NaveenRawatKentico** [brianvoe/slim-select#375] [Q&A] [answered]
Hi @brianvoe,

I am implementing SlimSelect first time and everything goes well except Events.

My Html looks like 

SlimSelect Initialization
<select id="divid" multiple>
        <option value="1">one</option>
        <option value="2">two</option>
        <option value="3">three</option>
        <option value="4">four</option>
    </select>
Html Select with id 'divid'
![image](https://user-images.githubusercontent.com/111181487/204838143-1496fa30-b220-4f95-8216-489ce04f1640.png)

<script src="https://cdnjs.cloudflare.com/ajax/libs/slim-select/1.27.1/slimselect.min.js"></script>
<link href="https://cdnjs.cloudflare.com/ajax/libs/slim-select/1.27.1/slimselect.min.css" rel="stylesheet" />
<script type="text/javascript">
        $(document).ready(function () {
            new SlimSelect({
                select: '#divid',
                events: {
                    afterChange: (newVal) => {
                        console.log(newVal);
                    }
                }
            });
        });
    </script>
    I am not getting anything in console. Any help?

**Comment by brianvoe** [comment, marked as answer]
Looks like you are on an older version of slim select. I setup a codepen for you to look at to get an idea of script imports.
https://codepen.io/brianvoe/pen/MWXqXXV

I am more curious about how you got to pull 1.27 when the docs all point to version 2.

**Reply by NaveenRawatKentico** [reply]
Thanks for your quick response.
When I visited few days ago to https://slimselectjs.com/, It was completely different UI and the instructions to use it.

It's working now but when I select from dropdown, every time it creates two same arrays.

![image](https://user-images.githubusercontent.com/111181487/204859830-7ef61d4a-fde2-4a2b-b2c8-80fbfc7d392e.png)


**Also, Can we get current instance of select inside events when we are having dynamic select ID**

var selectDB = new SlimSelect({
                            select: document.getElementById(i),
                            events: {
                                afterChange: (newVal) => {
                                   // current instance of select
                                }
                            }
                        });

**Comment by brianvoe** [comment]
Given that you need to pass in the instance of the select element, it would be best to grab it from your own establish selection.

**Reply by NaveenRawatKentico** [reply]
Hi @brianvoe 
What does this mean "your own establish selection" ?
What about the two instances created of selected item from dropdown (afterChange Event). Is it a known Issue ?
I also tried it in the codepen which you shared - https://codepen.io/brianvoe/pen/MWXqXXV.
I am using the latest version too.
